src/entity.ts

```diff
@@ -130,7 +150,48 @@ export namespace entity {
        * @name Int#value
        * @type {string}
        */
-      this.value = value.toString();
+      if (typeCastOptions) {
+        if (
+          typeCastOptions.integerTypeCastFunction &&
+          typeof typeCastOptions.integerTypeCastFunction !== 'function'
+        ) {
+          throw new Error(
+            `integerTypeCastFunction is not a function or was not provided.`
+          );
+        }
+        this.typeCastFunction = typeCastOptions.integerTypeCastFunction;
+        this.typeCastProperties = typeCastOptions.properties
+          ? arrify(typeCastOptions.properties)
+          : undefined;
+      }
+    }
+    // tslint:disable-next-line no-any
+    valueOf(): any {
+      let customCast = this.typeCastFunction ? true : false;
```

Comment:
It might be a bit more readable to ask exactly what we're wondering:

```js
let shouldCustomCast = typeof this.typeCastFunction === 'function';
```

Reply:
The first reason I thought to suggest this change was due to the `this.typeCastFunction ? true : false`, which is somewhat unusual-- that is coercing the type based on if it's not undefined/null. For readability, I would rather see it say exactly what we're wondering: "We should custom cast the integer if the user gave us a function", vs "We should custom cast the integer if the user gave us a non-null argument".

Reply:
Got it.
My line of thought was, since we already weeded out anything but a `function`, at this point the value of `this.typeCastFunction` can only be a `function`.
But for code readability purposes it makes sense.
Fixed it.